up42/workflow.py

```diff
@@ -44,12 +44,12 @@ class Workflow:
     def __init__(
         self,
         auth: Auth,
-        project_id: str,
         workflow_id: str,
+        project_id: Optional[str] = None,
         workflow_info: Optional[dict] = None,
     ):
         self.auth = auth
-        self.project_id = project_id
+        self.project_id = project_id if project_id is not None else self.auth.credentials_id
```

Comment:
I know this can be the opposite of your idea to add the project_id but if this is not included it would mean a breaking change on the client side. Please, correct me if I am wrong.

Reply:
Same apply to other classes.